### ISSUE TITLE
datasource by name api broke in Grafana 9.1.0

### ISSUE THREAD

**Issue by rrey** [grafana/grafana#53995]
**What happened**:

Integration tests of the Ansible Grafana collection broke with Grafana 9.1.0.
See details [here](https://github.com/ansible-collections/community.grafana/pull/271)

Investigating the issue, I found out that the API endpoint to [get datasource by name](https://grafana.com/docs/grafana/latest/developers/http_api/data_source/#get-a-single-data-source-by-name) is no more behaving properly.

Before 9.1.0, trying to get a datasource by name that do not exists resulted in the following error:
```
$ curl -u admin:admin http://127.0.0.1:3000/api/datasources/name/datasource%2FElastic
{"message":"Data source not found"}
```
Now it is returning:
```
$ curl -u admin:admin http://127.0.0.1:3000/api/datasources/name/datasource%2FElastic
{"accessErrorId":"ACE1254348339","message":"You'll need additional permissions to perform this action. Permissions needed: datasources:read","title":"Access denied"}
```
Which is incorrect.

**What you expected to happen**:

Proper error.

**How to reproduce it (as minimally and precisely as possible)**:

Start a fresh grafana 9.1.0 instance and run the following command:
```
curl -u admin:admin http://127.0.0.1:3000/api/datasources/name/datasource%2FElastic
```

(datasource `datasource/Elastic` do not exists and the response should reflect that)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.1.0
- Data source type & version: Any
- OS Grafana is installed on: docker image
- User OS & Browser: N/A
- Grafana plugins: None

**Comment by kalleep** [comment]
Hey, we merged a fix for this https://github.com/grafana/grafana/pull/54208. Going to check if we want to backport to 9.0.x and 9.1.x

**Comment by rrey** [comment]
Just tested with Grafana 9.1.2 and I have the same error:

```
$ curl -u admin:admin http://127.0.0.1:3000/api/datasources/name/datasource%2FElastic
{"accessErrorId":"ACE4471014249","message":"You'll need additional permissions to perform this action. Permissions needed: datasources:read","title":"Access denied"}
`
``

I re-tested with the latest 9.0.x (9.0.8) and I don't have any issue:
```
$ curl -u admin:admin http://127.0.0.1:3000/api/datasources/name/datasource%2FElastic
{"message":"Data source not found"}
```

**Comment by kalleep** [comment]
Added a backport to 9.1.x so the fix will be included in the next 9.1 release

**Comment by rrey** [comment]
my tests are no more failing with `9.1.7`, closing this issue.
